contracts/limitOrder/LimitOrderFeeVault.sol

```diff
@@ -0,0 +1,101 @@
+import { LimitOrderFeeVaultStorageV1 } from "../storage/LimitOrderFeeVaultStorage.sol";
+import { OwnerPausable } from "../base/OwnerPausable.sol";
+import { ReentrancyGuardUpgradeable } from "@openzeppelin/contracts-upgradeable/utils/ReentrancyGuardUpgradeable.sol";
+// solhint-disable-next-line max-line-length
+import { SafeERC20Upgradeable, IERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/SafeERC20Upgradeable.sol";
+
+contract LimitOrderFeeVault is
+    ILimitOrderFeeVault,
+    BlockContext,
+    ReentrancyGuardUpgradeable,
+    OwnerPausable,
+    LimitOrderFeeVaultStorageV1
+{
+    using AddressUpgradeable for address;
+    using PerpMath for int256;
+    using PerpMath for uint256;
+    using SignedSafeMathUpgradeable for int256;
+
+    modifier onlyLimitOrderBook() {
+        // LOFV_SMBLOB: Sender Must Be LimitOrderBook
+        require(_msgSender() == limitOrderBook, "LOFV_SMBLOB");
+        _;
+    }
+
+    function initialize(address rewardTokenArg, uint256 feeAmountArg) external initializer {
+        __OwnerPausable_init();
+        __ReentrancyGuard_init();
+
+        // LOFV_RTINC: RewardToken Is Not a Contract
+        require(rewardTokenArg.isContract(), "LOFV_RTINC");
+        rewardToken = rewardTokenArg;
+
+        // LOFV_FAMBGT0: FeeAmount Must Be Greater Than 0
+        require(feeAmountArg > 0, "LOFV_FAMBGT0");
+        feeAmount = feeAmountArg;
+    }
+
+    function setRewardToken(address rewardTokenArg) external onlyOwner {
+        // LOFV_RTINC: RewardToken Is Not a Contract
+        require(rewardTokenArg.isContract(), "LOFV_RTINC");
+        rewardToken = rewardTokenArg;
+        emit RewardTokenChanged(rewardTokenArg);
+    }
+
+    function setLimitOrderBook(address limitOrderBookArg) external onlyOwner {
+        // LOFV_LOBINC: LimitOrderBook Is Not a Contract
+        require(limitOrderBookArg.isContract(), "LOFV_LOBINC");
+        limitOrderBook = limitOrderBookArg;
+        emit LimitOrderBookChanged(limitOrderBookArg);
+    }
+
+    function setFeeAmount(uint256 feeAmountArg) external onlyOwner {
+        // LOFV_FAMBGT0: FeeAmount Must Be Greater Than 0
+        require(feeAmountArg > 0, "LOFV_FAMBGT0");
+        feeAmount = feeAmountArg;
+        emit FeeAmountChanged(feeAmountArg);
+    }
+
+    function disburse(address keeper, uint256 orderValue)
+        external
+        override
+        onlyLimitOrderBook
+        nonReentrant
+        returns (uint256)
+    {
+        // TODO: be aware of decimal issue when we use different reward token
+
+        // LOFV_NEBTD: not enough balance to disburse
+        require(IERC20Upgradeable(rewardToken).balanceOf(address(this)) >= feeAmount, "LOFV_NEBTD");
+
+        SafeERC20Upgradeable.safeTransfer(IERC20Upgradeable(rewardToken), keeper, feeAmount);
+
+        emit Disbursed(keeper, feeAmount);
+
+        return feeAmount;
+    }
+
+    // TODO: should we support multiple token to withdraw? or only support rewardToken?
```

Comment:
I think we could support one token at the beginning. `disburse()` only supports one token as well. We should align the behaviors of all the functions.